modules/gateway/peers.go

```diff
@@ -162,6 +163,40 @@ func (g *Gateway) acceptConn(conn net.Conn) {
 		return
 	}
 
+	// Read the peer's port that we can dial them back on. Peers older than v0.6.1
+	// will only be discovered by newer peers via the ShareNodes RPC.
+	var remoteAddr modules.NetAddress
+	if build.VersionCmp(remoteVersion, "0.6.1") >= 0 {
```

Comment:
this branch is larger enough that it might warrant its own function. e.g.
```go
switch {
case build.VersionCmp(remoteVersion, "0.6.1"):
    return g.addPeer(blah, blah, blah)
default:
    return g.addOldPeer(blah, blah, blah)
}
```
depends on how ugly the solution winds up being